File: src/main/kotlin/com/chernov/weather/config/LoggingAspectConfig.java
package com.chernov.weather.config;

import com.chernov.weather.config.LoggingAspect;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.context.annotation.EnableAspectJAutoProxy;
import org.springframework.core.env.Environment;

@Configuration
@EnableAspectJAutoProxy
public class LoggingAspectConfig {

    @Bean
    public LoggingAspect loggingAspect(Environment env) {
        return new LoggingAspect(env);
    }
}

File: src/main/kotlin/com/chernov/weather/config/LoggingAspectConfig.kt
package com.chernov.weather.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.context.annotation.EnableAspectJAutoProxy
import org.springframework.core.env.Environment

@Configuration
@EnableAspectJAutoProxy
class LoggingAspectConfig

@Bean
fun loggingAspect(env: Environment) = LoggingAspect(env)



